altsrc/map_input_source.go

```diff
@@ -32,11 +32,19 @@ func nestedVal(name string, tree map[interface{}]interface{}) (interface{}, bool
 			if !ok {
 				return nil, false
 			}
-			ctype, ok := child.(map[interface{}]interface{})
-			if !ok {
+
+			switch child := child.(type) {
+			case map[string]interface{}:
+				m := make(map[interface{}]interface{}, len(child))
```

Comment:
Bit of a nitpick, but I guess you could use `node = make(...` here. It's not a big deal though.

Reply:
Good point :slightly_smiling_face: